### PR TITLE
added messages about finishing shopping and ending products

### PR DESCRIPTION
**Pull request by Dyndyn**

### CODE REVIEW

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/StoragePageService.java [hunk 1]
```diff
@@ -6,13 +6,16 @@
...
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
 import org.apache.commons.collections.CollectionUtils;
 
 import java.sql.Timestamp;
 import java.text.SimpleDateFormat;
 import java.util.List;
+import java.util.Objects;
```
Comment (Dyndyn) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/StoragePageService.java:18]: MINOR Remove this unused import 'java.util.Objects'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/StoragePageService.java [hunk 2]
```diff
@@ -27,14 +30,14 @@
...
     private String storageUrl;
-
     @Value("${application.restShoppingListURL}")
     private String shoppingListURL;
 
     private final RestTemplate restTemplate;
     private final ShoppingListService shoppingListService;
     private final AnalyticsService analyticsService;
     private final MessageService messageService;
+    private final String productEnding = "storage.productEnding";
```
Comment (Dyndyn) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/StoragePageService.java:40]: MINOR Make this final field static too. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1170)

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/GoShoppingPageService.java [hunk 3]
```diff
@@ -25,13 +25,19 @@
 public class GoShoppingPageService {
     private static final Logger LOGGER = LogManager.getLogger(GoShoppingPageService.class);
     private final RestTemplate restTemplate;
+    private final CartService cartService;
+    private final MessageService messageService;
+    private static final String LOCALE_COOKIE = "myLocaleCookie";
```
Comment (Dyndyn) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/GoShoppingPageService.java:30]: MAJOR Remove this unused "LOCALE_COOKIE" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by Dyndyn** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/StoragePageService.java [hunk 4]
```diff
@@ -102,4 +102,26 @@ public void addProductToShoppingList(final int productId) {
             shoppingListService.addProductToShoppingList(productId);
         }
     }
+
+    public void reviewStorage(final String locale, final int userId) {
+        final List<Storage> storage = getStorages(userId);
+        storage.stream().filter(elem -> elem.getEndDate() != null)
+                .filter(elem -> elem.getEndDate().before(new Timestamp(System.currentTimeMillis() + 86400000l)))
```
Comment (kucherif) [inline MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/StoragePageService.java:109]: 86400000l - move to constants and give meaningful name

File: MVCapplication/src/main/webapp/WEB-INF/resources/js/emptyCart.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+                method: "POST"
+            });
+        }
+    }
+    else{
+        setTimeout(waitForElement, 250);
+    }
+}
+$(document).ready(function () {
+    waitForElement();
```
Comment (kucherif) [inline MVCapplication/src/main/webapp/WEB-INF/resources/js/emptyCart.js:15]: Could you please explain why do you need this?

Comment (Dyndyn) [reply MVCapplication/src/main/webapp/WEB-INF/resources/js/emptyCart.js:15]: this function waits for establishing WebSocket connection, and then make request to print message about finishing shopping

**Pull request merged**